docs/guide/state.md

```diff
@@ -0,0 +1,569 @@
+| AL_INVALID_ENUM      | The specified parameter is not valid. |
+| AL_INVALID_VALUE     | The specified data pointer is not valid. |
+| AL_INVALID_OPERATION | There is no current context. |
+
+##### Version Requirements
+OpenAL 1.0 or higher
+
+##### Remarks
+None
+
+##### See Also
+[alGetBoolean](#algetboolean), [alGetBooleanv](#algetbooleanv),
+[alGetDouble](#algetdouble), [alGetDoublev](#algetdoublev),
+[alGetFloat](#algetfloat), [alGetInteger](#algetinteger),
+[alGetIntegerv](#algetintegerv)
+
+#### alGetIntegerv
+##### Description
+This function retrieves an integer OpenAL state.
+
+```cpp
+void alGetIntegerv(
+    ALenum param,
+	ALint* data
+);
+```
+
+##### Parameters
+* param - the state to be queried:
+    + `AL_DOPPLER_FACTOR`
+	+ `AL_SPEED_OF_SOUND`
+	+ `AL_DISTANCE_MODEL`
+* data - A pointer to the location where the state will be stored
+
+##### Possible Error States
+| State                | Description |
+| -------------------- | ----------- |
+| AL_INVALID_ENUM      | The specified parameter is not valid. |
+| AL_INVALID_VALUE     | The specified data pointer is not valid. |
+| AL_INVALID_OPERATION | There is no current context. |
+
+##### Version Requirements
+OpenAL 1.0 or higher
+
+##### Remarks
+None
+
+##### See Also
+[alGetBoolean](#algetboolean), [alGetBooleanv](#algetbooleanv),
+[alGetDouble](#algetdouble), [alGetDoublev](#algetdoublev),
+[alGetFloat](#algetfloat), [alGetFloatv](#algetfloatv),
+[alGetInteger](#algetinteger)
+
+#### alGetString
+##### Description
+This function retrieves an OpenAL string property.
+
+```cpp
+const ALchar* alGetString(
+    ALenum param
+);
+```
+
+##### Parameters
+* param - the state to be queried:
+    + `AL_VENDOR`
+	+ `AL_VERSION`
+	+ `AL_RENDERER`
+	+ `AL_EXTENSIONS`
+
+##### Possible Error States
+| State                | Description |
+| -------------------- | ----------- |
+| AL_INVALID_ENUM      | The specified parameter is not valid. |
+
+##### Version Requirements
+OpenAL 1.0 or higher
+
+##### Remarks
+Returns a pointer to a null-terminated string.
+
+#### alDistanceModel
+##### Description
+This function selects the OpenAL distance model – `AL_INVERSE_DISTANCE`,
+`AL_INVERSE_DISTANCE_CLAMPED`, `AL_LINEAR_DISTANCE`,
+`AL_LINEAR_DISTANCE_CLAMPED`, `AL_EXPONENT_DISTANCE`,
+`AL_EXPONENT_DISTANCE_CLAMPED`, or `AL_NONE`.
+
+The `AL_INVERSE_DISTANCE` model works according to the following formula:
+```cpp
+gain = AL_REFERENCE_DISTANCE / (AL_REFERENCE_DISTANCE
+     + AL_ROLLOFF_FACTOR * (distance - AL_REFERENCE_DISTANCE));
+```
+
+The `AL_INVERSE_DISTANCE_CLAMPED` model works according to the following
+formula:
+```cpp
+distance = max(distance, AL_REFERENCE_DISTANCE);
+distance = min(distance, AL_MAX_DISTANCE);
+gain = AL_REFERENCE_DISTANCE / (AL_REFERENCE_DISTANCE
+     + AL_ROLLOFF_FACTOR * (distance - AL_REFERENCE_DISTANCE));
+```
+
+Here is a graph showing the inverse distance curve:
+![Inverse Distance Curve](images/figure-inverse-distance.jpg)

```

Comment:
This seems to be showing as a link, rather than an inline image. The same happens for the Linear and Exponential figures, which also have an incorrect caption (they both say "Inverse Distance Curve" as well).